relay-redis/src/real.rs

```diff
@@ -22,6 +22,7 @@ pub enum RedisError {
 
 enum ConnectionInner<'a> {
     Cluster(&'a mut redis::cluster::ClusterConnection),
+    MultiWrite(Vec<ConnectionInner<'a>>),
```

Comment:
If you split this into `MultiWrite { primary, other }` you don't have to expect/unwrap. You can also enforce that same constraint (it always requires a primary) at configuration level which means you never have to validate that constraint manually (although if you do it might require some Deserializer tricks/a custom impl).